### ISSUE TITLE
Can't preview unsaved page when parent has had children deleted

### ISSUE THREAD

**Issue by loicteixeira** [torchbox/wagtail#2599]
Seen with Python 3.4.3 & 3.5.1 and Wagtail 1.4.3

Steps to reproduce:
- Create a Page object, make and run the migrations.
- Open the admin.
- Create and publish a page under the root (let's call this page PP for Parent Page).
- Create and publish a couple pages under PP (let's call those pages CP1 and CP2 for Child Page X).
- Delete CP1.
- Create a page under PP. Don't save it, just click preview.
- Observe the error `ValidationError`, see stack trace below.

```
Internal Server Error: /admin/pages/add/core/articlepage/24/preview/
Traceback (most recent call last):
  File "/var/www/inner.kiwi/.pyenv/versions/inner.kiwi/lib/python3.5/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/var/www/inner.kiwi/.pyenv/versions/inner.kiwi/lib/python3.5/site-packages/django/views/decorators/cache.py", line 43, in _cache_controlled
    response = viewfunc(request, *args, **kw)
  File "/var/www/inner.kiwi/.pyenv/versions/inner.kiwi/lib/python3.5/site-packages/wagtail/wagtailadmin/decorators.py", line 22, in decorated_view
    return view_func(request, *args, **kwargs)
  File "/var/www/inner.kiwi/.pyenv/versions/inner.kiwi/lib/python3.5/site-packages/wagtail/wagtailadmin/views/pages.py", line 530, in preview_on_create
    page.full_clean()
  File "/var/www/inner.kiwi/.pyenv/versions/inner.kiwi/lib/python3.5/site-packages/wagtail/wagtailcore/models.py", line 436, in full_clean
    super(Page, self).full_clean(*args, **kwargs)
  File "/var/www/inner.kiwi/.pyenv/versions/inner.kiwi/lib/python3.5/site-packages/django/db/models/base.py", line 1171, in full_clean
    raise ValidationError(errors)
django.core.exceptions.ValidationError: {'path': ['Page with this Path already exists.']}
```

After some digging, it turns out that [previewing an unsaved page](https://github.com/torchbox/wagtail/blob/stable/1.4.x/wagtail/wagtailadmin/views/pages.py#L500) manually [set the path](https://github.com/torchbox/wagtail/blob/stable/1.4.x/wagtail/wagtailadmin/views/pages.py#L519-525) (but [previewing a saved page](https://github.com/torchbox/wagtail/blob/stable/1.4.x/wagtail/wagtailadmin/views/pages.py#L466) does not since the page already has a valid path).

The comment clearly state how tricky it is and that this is not working a 100% and I don't have time to look for a fix at the moment but hopefully, filling an issue will save a fellow developer some time :)

**Comment by loicteixeira** [comment]
A bit more info about paths (for another developer to look for a fix or me to remember about it when I have time to do so :o).

It uses treebeard where each path level is represented by a string of 4 characters. It's like an integer ID, but use base 36 (using `0-9a-z` range), so the first page on a node will have the path `0001`, the second page will be `0002`, the 10th page will be `000A`, the 37th page will be `0010` and so on.

So with wagtail, the root node is `0001`. The first page you create in the admin interface will be a child of the root node and have a full path of `00010001`. Children of that page will be `000100010001` and `000100010002` and so on.

Now when [the path is set manually](https://github.com/torchbox/wagtail/blob/stable/1.4.x/wagtail/wagtailadmin/views/pages.py#L519-525), it uses the number of children + 1 which is fine as long as no children has been deleted (the 3rd children will output a valid path `xxxxyyyy0003`).

However, like describe in the issue, if you create two pages, they will be given the paths `xxxxyyyy0001` and `xxxxyyyy0002`, then remove the first one. The number of children will be 1 again, therefore trying to create a path for the 2nd children, which will be `xxxxyyyy0002` and which unfortunately already exists.

The thing is, treebeard des not seem to rely on the number of children and handle that case without issues. This is probably what we should try to accomplish.

**Comment by gasman** [comment]
Bug confirmed on latest master. Thanks for the in-depth investigation!

**Comment by gasman** [comment]
Fix: #2616
